cli/config.go

```diff
@@ -56,27 +100,41 @@ type Config struct {
 	Client client.Config `mapstructure:"client"`
 }
 
-func loadConfig(cmd *cobra.Command) (Config, error) {
-	var opts []config.LoaderOption
-
-	cfgFile, _ := cmd.Flags().GetString(configFlag)
-	if cfgFile != "" {
-		opts = append(opts, config.WithFile(cfgFile))
-	} else {
-		opts = append(opts,
-			config.WithPath("./"),
-			config.WithName("compass"),
-			config.WithEnvKeyReplacer(".", "_"),
-			config.WithEnvPrefix("COMPASS"),
-		)
+func LoadConfig() (*Config, error) {
+	var cfg Config
+	err := cmdx.SetConfig("compass").Load(&cfg)
+	if err != nil {
+		if errors.As(err, &config.ConfigFileNotFoundError{}) {
+			return LoadFromCurrentDir()
+		}
+		return &cfg, err
 	}
+	return &cfg, nil
+}
 
+func LoadFromCurrentDir() (*Config, error) {
 	var cfg Config
+	var opts []config.LoaderOption
+
+	opts = append(opts,
+		config.WithPath("./"),
+		config.WithName("compass.yaml"),
+		config.WithEnvKeyReplacer(".", "_"),
+		config.WithEnvPrefix("COMPASS"),
+	)
+
 	if err := config.NewLoader(opts...).Load(&cfg); err != nil {
 		if errors.As(err, &config.ConfigFileNotFoundError{}) {
-			return cfg, nil
+			return &cfg, ErrConfigNotFound
 		}
-		return cfg, err
+		return &cfg, err
 	}
-	return cfg, nil
+	return &cfg, nil
+}
```

Comment:
The current implementation looks like we give lower priority to the config file in the current directory, I feel we should do the opposite, and try to load from the current first if fails, we can fall back to the rest of the pre-defined directories.